### PR TITLE
MLPAB-1273 - batch replicate objects to Sirius

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
# Purpose

Automate the creation of s3 batch replication jobs using manifest generation from s3 bucket replication configuration.

Fixes MLPAB-1273

## Approach

**Lambda**
- write python script to 
  - pull job configuration from AWS SSM Parameter Store
  - create a S3 batch replication job
  - generate a manifest using the S3 bucket's replication configuration
- create lambda function Dockerfile

**Github Actions workflows**
- build and push lambda image on pipeline using a github actions matrix
- trigger rebuild (full workflow) when lambda code changes
- add a timeout for account level terraform plan steps so that it waits a while to acquire state lock, rather than failing immediately (should result in fewer workflows failing when a merge to main happens)

**Terraform**
- update IAM policy on replicate role to allow batch replication
- create a lambda function
  - add permissions for lambda function to 
    - access SSM Parameter Store
    - create a S3 batch job
    - pass a role to the S3 batch job
- allow operator role to execute lambda function

## Learning

- https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication.html#batch-replication-scenario
- https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3control.html
- https://docs.aws.amazon.com/AmazonS3/latest/userguide/batch-ops-create-job.html
- https://boto3.amazonaws.com/v1/documentation/api/1.9.185/reference/services/ssm.html#SSM.Client.get_parameter
- https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs
- https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_passrole.html

### CODE REVIEW

File: lambda/create_s3_replication_job/src/main.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                    'SSES3': {}
+                },
+                'ManifestFormat': 'S3InventoryReport_CSV_20211130'
+            },
+            'Filter': {
+                'EligibleForReplication': True,
+                'ObjectReplicationStatuses': [
+                    'FAILED',
+                    'NONE'
+                ]
```
Comment (hawx) [inline lambda/create_s3_replication_job/src/main.py:83]: should this also filter for things that have passed the virus scan?

Comment (andrewpearce-digital) [reply lambda/create_s3_replication_job/src/main.py:83]: The replication configuration on the bucket does that filter.

Comment (andrewpearce-digital) [reply lambda/create_s3_replication_job/src/main.py:83]: I'm going to check on that actually before requesting a review again

Comment (andrewpearce-digital) [reply lambda/create_s3_replication_job/src/main.py:83]: tested, and it works fine. See ticket for test results

**Review by hawx — APPROVED** [review]

**Review by hawx — APPROVED** [review]

**Pull request merged**